src/CSTools/Tools/parser.cs

```diff
@@ -1504,7 +1504,7 @@ protected bool Error(ref ParseStackEntry top, string str)
 								newtop = pe.m_action.Action(this); // before we change the stack
 							m_ungot = top.m_value;
 							Pop(ref top,((ParserReduce)pe).m_depth,er);
-							newtop.pos = top.m_value.pos;
+							newtop!.pos = top.m_value.pos;
```

Comment:
wont this throw if `pe.m_action` is null?

Reply:
Yes, it will. The way I interpret this section is, "if pe is non-null, so is pe.m_action"